cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -562,7 +562,51 @@ impl Lockfile {
             .collect::<Result<Vec<_>, LockedManifestError>>()
     }
 
-    /// Produce a lockfile for a given manifest.
+    /// Merge included environments, resolve the merged manifest, and return the resulting lockfile
+    ///
+    /// Already resolved packages will not be re-resolved,
+    /// and already fetched includes will not be re-fetched.
+    pub async fn lock_manifest(
+        manifest: &Manifest,
+        seed_lockfile: Option<&Lockfile>,
+        client: &impl catalog::ClientTrait,
+        installable_locker: &impl InstallableLocker,
+    ) -> Result<Lockfile, LockedManifestError> {
```

Comment:
I feel a bit uneasy with pulling the composition (esp. the fetching of environments) into the locking/lockfile layer.

To me an exemplary call stack 

```
PathEnvironment
  > Lockfile
    > PathEnvironment  ( include )
      > Lockfile ( via ensure_locked )
```

seems like a violation of separation of concerns to me compared to

```
PathEnvironment
  > PathEnvironment ( include )
    > Lockfile ( via ensure_locked )
  > Lockfile    
```

Also considering that the error type for locking now would also need to contain EnvironmentError which appears equally backwards.


At least the fetching of dependencies IMO belongs to the higher level environment domain. the constituents are part of the lock, but locking can work with a `Compose` input with minimal changes.

Reply:
See https://flox-dev.slack.com/archives/C05P6A5J6U8/p1740670880622389

We could consider more cleanly separating merging and resolving functionality (e.g. `lock(Resolver, Merger) -> Lockfile`), but that doesn't seem necessary at the moment